core/src/commitment.rs

```diff
@@ -286,6 +328,23 @@ impl AggregateCommitmentService {
                     break;
                 }
             }
+            let mut insert_amount = lamports;
```

Comment:
Hmmm, would it be simpler if instead of a `BTreeMap` we just inserted all the (root, stake) into an unsorted vector `V`. Then here: https://github.com/solana-labs/solana/pull/9640/files#diff-17e339edccbb98467b9ae80777ce4e31R186, we do `V.sort_by(|a, b| a.0.cmp(b.0).reverse())`. We then loop through `v in V` summing the stake `v.1` until we see something >2/3 and that's the greatest confirmed root

Reply:
Yes, would definitely make the insert better!

Reply:
✔️